src/wasm/wasm-binary.cpp

```diff
@@ -7981,6 +7981,52 @@ static void readResumeTable(WasmBinaryReader* reader, ResumeType* curr) {
   }
 }
 
+template<typename ResumeOrResumeThrow>
+static void computeResumeTableSentTypes(Module& wasm,
+                                        ResumeOrResumeThrow* curr) {
+  static_assert(std::is_base_of<Resume, ResumeOrResumeThrow>::value ||
+                std::is_base_of<ResumeThrow, ResumeOrResumeThrow>::value);
+  assert(curr->handlerBlocks.size() == curr->handlerTags.size());
+
+  // Let $tag be a tag with type [tgp*] -> [tgr*]. Let $ct be a continuation
+  // type (cont $ft), where $ft is [ctp*] -> [ctr*]. Then an instruction
+  // (resume $ct ... (tag $tag $block) ... ) causes $block to receive values
+  // of the following types when suspending to $tag: tgp* (ref $ct') where ct'
+  // = (cont $ft') and ft' = [tgr*] -> [ctr*].
+  //
+  curr->sentTypes.reserve(curr->handlerTags.size());
+  auto contSig = curr->contType.getContinuation().type.getSignature();
+  auto& ctrs = contSig.params;
+  for (Index i = 0; i < curr->handlerTags.size(); i++) {
+    Type sentType;
+    if (curr->handlerBlocks[i].isNull()) {
+      sentType = Type::none;
+    } else {
+      auto& tag = curr->handlerTags[i];
+      auto& tagSig = wasm.getTag(tag)->sig;
+
+      auto& tgps = tagSig.params;
+      auto& tgrs = tagSig.results;
+
+      HeapType ftPrime{Signature(tgrs, ctrs)};
+      HeapType ctPrime{Continuation(ftPrime)};
+      Type ctPrimeRef(ctPrime, Nullability::NonNullable);
```

Comment:
Shouldn't this be looking up the types based on the labels now?

Reply:
Is your point that we can simply read the type of the continuation off the (corresponding branch target) label type? I think so yes, it must have been given statically in the source code. It is not entirely clear to me what the purpose of `sentTypes` is (I suppose it is some caching mechanism).